### ISSUE TITLE
EF Core 6.0 temporal tables - extra migrations generated when no change

### ISSUE THREAD

**Issue by slubowsky** [dotnet/efcore#27911]
When add temporal table and set history table schema, if add migration again in future, extra incorrect migrations are created.

In DbContext.OnModelCreating made many existing tables temporal by adding .IsTemporal()
```
modelBuilder.Entity<MyEntity>()
               .ToTable("MyEntity", b => b.IsTemporal());
```

Set history schema and temporal column names on all like this
```
foreach (var entityType in modelBuilder.Model.GetEntityTypes())
{
    if (entityType.IsTemporal())
    {
       entityType.SetHistoryTableSchema("history");
       entityType.SetPeriodStartPropertyName("ValidFrom");
       entityType.SetPeriodEndPropertyName("ValidTo");
    }
}
```
Generated and applied migrations to DB. All seems well. Make additional changes and generate new migrations. Get an incorrect migration for each temporal table in addition to the expected migrations for the real changes:
```
migrationBuilder.AlterTable(
      name: "MyEntity")
      .Annotation("SqlServer:IsTemporal", true)
      .Annotation("SqlServer:TemporalHistoryTableName", "MyEntityHistory")
      .Annotation("SqlServer:TemporalHistoryTableSchema", "history")
      .Annotation("SqlServer:TemporalPeriodEndColumnName", "ValidTo")
      .Annotation("SqlServer:TemporalPeriodStartColumnName", "ValidFrom")
      .OldAnnotation("SqlServer:IsTemporal", true)
      .OldAnnotation("SqlServer:TemporalHistoryTableName", "MyEntityHistory")
      .OldAnnotation("SqlServer:TemporalHistoryTableSchema", null)
      .OldAnnotation("SqlServer:TemporalPeriodEndColumnName", "ValidTo")
      .OldAnnotation("SqlServer:TemporalPeriodStartColumnName", "ValidFrom");
```

Edit - deleted all existing migrations and just generated a first one immediately followed by another. Same incorrect behavior. Incorrect migrations included in second for all temporal tables despite no changes having been made.

EF Core version: 6.0.4
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: NET 6.0
Operating system: Windows 11
IDE: Visual Studio 2022 17.1.6

**Comment by maumar** [comment]
dupe of #26676

@slubowsky as a workaround on 6.0.4 you can use:


```cs
	protected override void OnModelCreating(ModelBuilder modelBuilder)
	{
		modelBuilder.Entity<MyEntity>().ToTable(b => b.IsTemporal(ttb => ttb.UseHistoryTable("history", "MyEntityHistory")));

                foreach (var entityType in modelBuilder.Model.GetEntityTypes())
                {
                    if (entityType.IsTemporal())
                    {
                        entityType.SetPeriodStartPropertyName("ValidFrom");
                        entityType.SetPeriodEndPropertyName("ValidTo");
                    }
                }
        }

```

**Comment by slubowsky** [comment]
@maumar ok thanks. I see its already marked as fixed for 7.0 (and maybe a 6.x release?). Until it hits, I think Ill just continue to manually remove the unwanted migrations rather then add that extra code to the many tables since it soon wont be needed.

**Comment by maumar** [comment]
@slubowsky it's fixed in current 7.0 preview but we rejected it for 6.x patch.
